src/corehost/src/tpafile.cpp

```diff
@@ -8,8 +8,28 @@
 #include "tpafile.h"
 #include "utils.h"
 
-bool read_field(pal::string_t line, int& offset, pal::string_t& value_recv)
+bool tpaentry_t::to_full_path(const pal::string_t& base, pal::string_t* str) const
 {
+    pal::string_t& candidate = *str;
+
+    candidate.reserve(base.length() +
+        library_name.length() +
+        library_version.length() +
+        relative_path.length() + 3);
+
+    candidate.assign(base);
+    append_path(candidate, library_name.c_str());
+    append_path(candidate, library_version.c_str());
+    append_path(candidate, relative_path.c_str());
+    
+    return pal::file_exists(*str);
+}
+
+bool read_field(const pal::string_t& line, pal::char_t* buf, unsigned* ofs, pal::string_t* field)
```

Comment:
I'm not sure what the best style is but I generally prefer to use `unsigned int` or `size_t` rather than `unsigned` to be a little clearer.